sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1430,9 +1430,13 @@ object Limit {
 /**
  * A global (coordinated) limit. This operator can emit at most `limitExpr` number in total.
  *
+ * We can not make it inherit [[OrderPreservingUnaryNode]] due to the different strategy of
+ * physical plan. The output ordering of child will be broken if a shuffle exchange comes in
+ * between the child and global limit.
```

Comment:
```suggestion
 * between the child and global limit, due to the fact that shuffle reader fetches blocks in random order.
```

Reply:
That said, we can bring this back in the future if our shuffle reader fetches blocks in sequence.